internal/notification-worker/worker.go

```diff
@@ -50,36 +50,53 @@ func (w *Worker) CloseResources() {
 
 func (w *Worker) checkInvoices() {
 	ctx := goContext.Background()
-	// Get invoices parties were not notified yet
-	invoices, err := w.db.GetNotNotifiedInvoices(ctx, w.config.BatchSize)
-	if err != nil {
-		return
-	}
 
-	if len(invoices) == 0 {
-		context.GetLogger(ctx).Info("worker.checkInvoices.noNewInvoices")
-		return
-	}
+	_ = w.db.RunInTransaction(ctx, func(ctx goContext.Context) error {
```

Comment:
I'm not sure such broad transaction is good idea, I think transaction can be committed right after update of rows to sending